### PR TITLE
Source generator document

### PR DESCRIPTION
**Pull request by kchinburarat**
Adding Agoda.IoC.Generator usage document :)

### CODE REVIEW

File: Generator.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+<PackageReference Include="Agoda.Ioc.Generator" Version="1.1.128"  PrivateAssets="all" />
+```
+
+## .NET CLI
+
+``` powershell
+dotnet add package Agoda.Ioc.Generator --version 1.1.128
+```
+
+**Note:** if you are using the .NET CLI. Please open `*.csproj` and add PrivateAssets="all" on PackageReference Include="Agoda.Ioc.Generator" Version="1.1.128" />
```
Comment (dicko2) [inline Generator.md]: You should be able to define this in the csproj file so the cli tool will do automatically

File: Generator.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+## .NET CLI
+
+``` powershell
+dotnet add package Agoda.Ioc.Generator --version 1.1.128
+```
+
+**Note:** if you are using the .NET CLI. Please open `*.csproj` and add PrivateAssets="all" on PackageReference Include="Agoda.Ioc.Generator" Version="1.1.128" /> 
+
+**Note:** Please install the "Microsoft.Extensions.DependencyInjection.Abstractions" package if your project is not an asp.net project.
```
Comment (dicko2) [inline Generator.md]: This should be a depenecy of our library so this is automatic

File: Generator.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+```
+It can also be used to register multiple instances
+
+``` csharp
+[RegisterSingleton(For = typeof(IMultipleAttributes1))]
+[RegisterSingleton(For = typeof(IMultipleAttributes2))]
+public class MultipleAttributes : IMultipleAttributes1, IMultipleAttributes2 {}
+```
+
+Additionally, it may be used to assign an attribute order and register many instances as a collection.
```
Comment (dicko2) [inline Generator.md:143]: Is this existing functionality or new?

I think if its existing we should use the same docs to describe both.

If it's new we should update the original library to do the same then again only have one doc that describes functionality

File: Generator.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.RegisterFromAgodaExample(); // Here 😃 
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
```
Comment (dicko2) [inline Generator.md]: I would remove all these other lines like swagger and controllers. Just have the container build and the commands needed for this library. Or it may confuse some people

File: Generator.md (hunk 1 above)
Comment (kchinburarat) [reply Generator.md]: I don't understand this.Could you please help explain more.

File: Generator.md (hunk 2 above)
Comment (kchinburarat) [reply Generator.md]: Added

File: Generator.md (hunk 3 above)
Comment (kchinburarat) [reply Generator.md:143]: This feature already exist so i will have to add on both

File: Generator.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
+# Installation
+
+## Add the NuGet Package to your project
+
+All you need to do, to install Agoda.Ioc.Generator is to add a NuGet reference pointing to the package Agoda.Ioc.Generator.
+
+## PackageReference
+
+``` xml
+<PackageReference Include="Agoda.Ioc.Generator" Version="1.1.128"   />
```
Comment (szaboopeeter) [inline Generator.md]: Maybe we shouldn't mention explicit versions in the documentation?

File: src/Generator/Agoda.IoC.Generator/Agoda.IoC.Generator.csproj [hunk 6]
```diff
@@ -17,26 +17,23 @@
...
 		<Description>Share dotnet C# IoC source generator, used at Agoda.</Description>
-
 	</PropertyGroup>
-
 	<ItemGroup>
 		<PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" PrivateAssets="all" />
 		<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.5.0" PrivateAssets="all" />
-	</ItemGroup>
-
+		<PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.1.0" />
```
Comment (szaboopeeter) [inline src/Generator/Agoda.IoC.Generator/Agoda.IoC.Generator.csproj]: Why are these needed for a documentation change?

File: Generator.md (hunk 4 above)
Comment (kchinburarat) [reply Generator.md]: removed

File: Generator.md (hunk 5 above)
Comment (kchinburarat) [reply Generator.md]: Fixed

File: src/Generator/Agoda.IoC.Generator/Agoda.IoC.Generator.csproj (hunk 6 above)
Comment (kchinburarat) [reply src/Generator/Agoda.IoC.Generator/Agoda.IoC.Generator.csproj]: removed

**Review by szaboopeeter — APPROVED** [review]
